README.md

```diff
@@ -1,8 +1,129 @@
-# caramel.cpp
+# Caramel: A fast read-only key-value store
 
-This repository provides a C++ implementation of the [CARAMEL data structure](https://arxiv.org/pdf/2305.16545.pdf), a space-efficient read-only retrieval table with applications to data management problems in recommender systems and computational biology. 
+This repository provides a C++ implementation of the [CARAMEL data structure](https://arxiv.org/pdf/2305.16545.pdf), a space-efficient read-only retrieval table with applications to data management problems in recommender systems, computational biology, and more.
 
-## Development Scripts
+## What is CARAMEL?
+
+CARAMEL (Compressed Array Representation And More Efficient Lookups) is a data structure built around compressed static functions (CSFs). For a solid introduction to CSFs, see the excellent series of papers by Sebastiano Vigna's group. We draw heavy inspiration from their [Java implementation in Sux4J](https://sux4j.di.unimi.it).
+
+**Why caramel?** Compared to Sux4J, we offer about 25% faster construction, a much smaller and simpler codebase (> 10k vs 2k LoC for the core implementation), and easy Python bindings. We support arbitrary data types for keys and values, enabling more flexibility for applications (the original implementation supports only string keys and integer values). We also have some algorithmic improvements to make CSF-backed tables practical for real applications. For example, we have optimal-size Bloom prefilters for situations when one value is very common. We also have value-permutation methods to preprocess 2D arrays into a more compressible format, as well as tricks to implement ragged arrays and other lookup primitives using CSFs. Our paper describes these tricks and applications.
+
+### Minimal API Example
+
+You can use our Python API like this.
+
+```python
+import caramel
+keys = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J"]
+values = [0, 0, 0, 1, 2, 0, 1, 4, 0, 1]
+# Constructing
+csf = caramel.CSF(keys, values)
+# Querying
+csf.query("D")  # Returns 1
+# Saving
+csf.save("map.csf")
+# Loading
+csf = caramel.load("map.csf")
+```
+
+## Getting Started
+The recommended way to interact with our library is through the Python bindings, though we also offer a C++ API. 
+
+### Python Installation
+To install the Python bindings, first clone the repository and all of our dependencies.
+
+```bash
+git clone --recurse-submodules git@github.com:detorresramos/caramel.cpp.git
+cd caramel.cpp
+```
+
+Next, run our build script:
+```bash
+python3 bin/build.py
```

Comment:
I can add some openmp install instructions. Do you happen to remember what you did when first installing? Was it just a brew install?

Reply:
Did not need anything special, probably already had it all on my machine?

If we really wanna be thorough, we could do a clean install and document whatever we needed for that. I don't have bandwidth for that right now though.

I also don't know if >= python3.6 and >= c++20 are strictly required or not.